js_modules/dagster-ui/packages/ui-core/src/app/UserSettingsDialog/UserSettingsDialog.tsx

```diff
@@ -61,7 +63,14 @@ const UserSettingsDialogContent = ({onClose, visibleFlags}: DialogContentProps)
   });
 
   const toggleFlag = (flag: FeatureFlag) => {
-    setFlags(flags.includes(flag) ? flags.filter((f) => f !== flag) : [...flags, flag]);
+    const flagSet = new Set(flags);
+    trackEvent('feature-flag', {flag, enabled: !flagSet.has(flag)});
```

Comment:
Hmm it'd be nice to also add this information to the view events here https://github.com/dagster-io/internal/blob/a1d9f622815fb70fae50920efafd2e93e376d6c9/dagster-cloud/js_modules/app-cloud/src/app/CloudAnalyticsProvider.tsx#L46

Also incorporating it into datadog might be useful to so that we could look at errors with the flag on vs off.

Reply:
I.e., include the full flag state in the pageview events?

Reply:
yeah